cocos/particle/renderer/particle-system-renderer-data.ts

```diff
@@ -203,11 +203,24 @@ export default class ParticleSystemRenderer {
 
     private _particleSystem: any = null!; // ParticleSystem
 
+    create (ps: any) {
+        if (this._particleSystem === null) {
+            this._particleSystem = ps;
+            const useGPU = this._useGPU && isSupportGPUParticle();
+            this._particleSystem.processor = useGPU ? new ParticleSystemRendererGPU(this) : new ParticleSystemRendererCPU(this);
+            this._particleSystem.processor.onInit(this._particleSystem);
```

Comment:
If particle is not enabled or node not activated in editor, this process will create more useless memory.
We need to control things more strictly, I think it's enough to assign `this._particleSystem = ps;`.

Reply:
create (ps: any) {
        if (this._particleSystem === null) {
            this._particleSystem = ps;
        }
        else if (this._particleSystem !== ps) {
            errorID(...); // Message: ParticleSystem cannot be created with different particle system
        }
    }